pkg/localmetrics/localmetrics.go

```diff
@@ -294,7 +295,11 @@ func (c *MetricsCollector) AddAccountReuseCleanupFailure() {
 
 // SetReconcileDuration describes the time it takes for the operator to complete a single reconcile loop
 func (c *MetricsCollector) SetReconcileDuration(controller string, duration float64, err error) {
-	c.reconcileDuration.WithLabelValues(controller, strconv.FormatBool(err != nil)).Observe(duration)
+	awsErrorCode := ""
+	if aerr, ok := err.(awserr.Error); ok {
```

Comment:
Summary of offline discussion:
- This is goodness. Keep it.
- But also see if you can add the dimension to the [`apiCallDuration` metric](https://github.com/openshift/aws-account-operator/pull/487/files#diff-2afe99d72c5464037a873b976d1699b8R155) by passing an `error` arg to [`AddAPICall`](https://github.com/openshift/aws-account-operator/pull/487/files#diff-2afe99d72c5464037a873b976d1699b8R310) based on the `Error` field of the AWS `request` object available to our [callback](https://github.com/openshift/aws-account-operator/blob/48be5995045f04924e28ada9c5f465274d26fafd/pkg/awsclient/client.go#L380)